awq/utils/calib_data.py

```diff
@@ -1,12 +1,19 @@
+from typing import List, Union
 import torch
 import logging
 from datasets import load_dataset
 
-def get_calib_dataset(data="pileval", tokenizer=None, n_samples=512, block_size=512):
-    if data == "pileval":
-        dataset = load_dataset("mit-han-lab/pile-val-backup", split="validation")
+def get_calib_dataset(data: Union[str, List[str]] = "pileval", tokenizer=None, n_samples=512, block_size=512):
+    if isinstance(data, str):
+        if data == "pileval":
+            dataset = load_dataset("mit-han-lab/pile-val-backup", split="validation")
+        else:
+            raise NotImplementedError
```

Comment:
Should work. Might want to find a way define the split instead of defaulting to train, though.
```suggestion
            dataset = load_dataset(data, split="train")
```

Reply:
The defaulting to `train` might be solved by adding a kwarg with that defaults to `validation` which could be used in L9 and L11.

Reply:
I agree that we need to not raise an exception here. Instead, we should try to load the dataset by the actual string that was passed and load the split by another variable passed. We could default to the validation split as this should be a small enough dataset for calibration yet scientifically sound enough since we would use the test split to measure perplexity.